src/Containers/NutritionFact/NutritionFact.tsx

```diff
@@ -0,0 +1,296 @@
+import React,{useCallback,useMemo,useRef,forwardRef,useEffect,useState} from 'react'
+import styled from 'styled-components'
+import { Mixins } from '../../Utils'
+import {
+    Main,
+    MainInterface,
+} from '../../Utils/BaseStyles';
+import Theme from '../../Themes/ThemeTemplate'
+import {Input} from '../../Inputs/Input/Input'
+
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+
+interface IAdditionalProps{
+    ref?:boolean;
+    delay?:boolean;
+}
+
+interface IEntries{
+    type:string;
+    entry:IEntryProps&IAdditionalProps|IHeadingEntryProps&IAdditionalProps;
+}
+
+export interface INutritionFactProps{
+    entries:IEntries[];
+    editMode?:boolean;
+}
+
+export const NutritionFact:React.FC<INutritionFactProps>=({entries,editMode}):React.ReactElement=>
+{
+    const mainContainerRef=useRef<HTMLDivElement>(null)
+
+    const containerRef=useRef<HTMLDivElement>(null)
+
+    const delayLabel=useRef<string[]>([])
+
+    const infoRef=useRef<IInfo|null>(null)
+
+    /**
+     * renders an entry of type entry
+     * @param entry {IEntryProps} - the entry to render
+     * @returns {React.ReactElement} the rendered entry
+     */
+    const renderEntry=useCallback((entry:IEntryProps&IAdditionalProps):React.ReactElement=>{
+        const {label,...rest}=entry
+        return <Entry key={label} label={label} {...rest} margin='0 0 2px' padding='0 0 1px' editMode={editMode} />
```

Comment:
Won't the padding always be overrided by the padding. Should ...rest be at the end :)

Reply:
entry is of type IEntryProps&IAdditionalProps which has no padding in it, so {...rest} contain no padding and will not be overwrited. In this case I this case I think it is not necessary {...rest} to be at the end 😉